manager/controllers/app/fybrikmodule_controller.go

```diff
@@ -107,7 +107,8 @@ func ValidateFybrikModule(module *fapp.FybrikModule, taxonomyFile string) error
 	var allErrs []*field.Error
 
 	// Convert Fybrik module Go struct to JSON
-	moduleJSON, err := json.Marshal(module)
+	spec := module.Spec.DeepCopy()
+	moduleJSON, err := json.Marshal(spec)
```

Comment:
Why do you need a deep copy here? json.Marshal(&module.Spec) should be enough, you are not using the saved pointer, just a list of bytes...